overlord/snapstate/aliases.go

```diff
@@ -187,11 +187,13 @@ func (m *SnapManager) doAlias(t *state.Task, _ *tomb.Tomb) error {
 		}
 		switch newStatus {
 		case "enabled":
-			err := checkAliasConflict(st, snapName, alias)
-			if err != nil {
-				return err
+			if aliasStatuses[alias] != "auto" {
+				err := checkAliasConflict(st, snapName, alias)
```

Comment:
I was slightly confused first why no conflict checking is done for auto mode. I see that this is done below, maybe a small comment here that its just handled in a different place?